docs/RelationalData.md

```diff
@@ -0,0 +1,197 @@
+
+## Relationship description
+There can be any number of relationships within each profile. Sub profiles can also contain relationships.
+
+A relationship description can have the following properties
+
+| property | type | requirement | description |
+| ---- | ---- | ---- | ---- |
+| `name` | string | required | The name of the relationship, this is the name of the property in which the related data will be embedded |
+| `description` | string | optional | A description for the relationship, this is for documentation only - it has no bearing on the generation process | 
+| `profileFile` | string | optional* | A relative path to a profile file, that describes the data that should be generated within this relationship |
+| `profile` | object | optional* | A profile that describes the data that should be generated within this relationship |
+| `extents` | array of `extent` | optional | The _min_ and _max_ number of records within a one-to-many relationship, if absent will be treated as a one-to-one relationship |
+
+For example:
+```
+{
+  ...
+  "relationships": [ 
+    {
+      "name": "name",
+      "profileFile": "profile-filename",
+      "extents": [
+          { "field": "min", "equalTo": 1 },
+          { "field": "max", "equalTo": 3 }
+      ],
+    }
+  ]
+}
+```
+
+\* One of `profile` or `profileFile` must be supplied, `profile` takes precedence over `profileFile`.
+
+**NOTE**: If `profileFile` is used, it is possible to create a circular/recursive 'dependency' on profiles. No current strategy has been implemented to prevent/detect this, however it should be simple to integrate.
+
+An `extent` is a description of the limit on the number of rows that must be produced, it is described as a regular - numeric - constraint, but for a virtual field called `min` or `max`, as in the example above. Any constraint that results in a integer value can be used.
+
+**If any extents are supplied** then a constraint for the `max` extent must be supplied. The `min` extent is 0 by default, but can be overridden with any &gt;= 0 value. Conditional constraints are supported.
+
+For one-to-many relationships a random number of records will be produced between `min` and `max`. Other strategies can be introduced in the fullness of time, this is the only approach at present.
+
+## Example profile:
+
+```
+{
+  "fields": [
+    {
+      "name": "shortName",
+      "type": "faker.name.firstName",
+      "nullable": false
+    }
+  ],
+  "constraints": [
+    {
+      "field": "shortName",
+      "shorterThan": 6
+    },
+    {
+      "field": "shortName",
+      "matchingRegex": "J.*"
+    }
+  ],
+  "relationships": 
+  [ 
+    {
+        "name": "dependants",
+        "description": "presence of min/max indicates that it is a collection (one-to-many)",
+        "profileFile": "dependants.profile.json",
+        "extents": [
+            { "field": "min", "equalTo": 0 },
```

Comment:
The generator allows for a range to be narrowed but never widened. Therefore the range starts off as `0..` then is adapted based on the conditions here

Reply:
Ah I see, this is a little confusing to me based on other constraints where specifying equalTo 0 and equalTo 1 is a contradiction. I'm not sure how this could be reworded to make more sense